ENIGMAsystem/SHELL/Graphics_Systems/General/GSmodel.cpp

```diff
@@ -15,15 +17,651 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
+#include "GSmodel_impl.h"
 #include "GSmodel.h"
+#include "GSvertex.h"
+#include "GSvertex_impl.h"
+#include "GSmatrix.h"
+#include "GStextures.h"
 #include "GSprimitives.h"
 
+#include "Universal_System/fileio.h"
+
 #include <math.h>
 
 using namespace std;
 
+namespace enigma {
+
+vector<Model*> models;
+extern unsigned char currentcolor[4];
+
+unsigned int split(const std::string &txt, std::vector<std::string> &strs, char ch)
+{
+	size_t pos = txt.find( ch );
+	size_t initialPos = 0;
+	strs.clear();
+
+	// Decompose statement
+	while( pos != std::string::npos ) {
+		strs.push_back( txt.substr( initialPos, pos - initialPos + 1 ) );
+		initialPos = pos + 1;
+
+		pos = txt.find( ch, initialPos );
+	}
+
+	// Add the last one
+	strs.push_back( txt.substr( initialPos, std::min( pos, txt.size() ) - initialPos + 1 ) );
+
+	return strs.size();
+}
+
+//split a string and convert to float
+vector<float> float_split(const string& str, const char& ch) {
+  string next;
+  vector<float> result;
+
+  for (string::const_iterator it = str.begin(); it != str.end(); it++)
+  {
+    if (*it == ch) {
+      if (!next.empty()) {
+        result.push_back(atof(next.c_str()));
+        next.clear();
+      }
+    } else {
+      next += *it;
+    }
+  }
+  if (!next.empty())
+    result.push_back(atof(next.c_str()));
+  return result;
+}
+
+//obj model parsing functions
+void string_parse(string *s) {
```

Comment:
It's not going to work now I don't think. I should have started by moving one of the original model sources first and then did the changes. It's not really worth it because this is otherwise an entirely new file. Correct me if I'm wrong.